Gems/Camera/Code/Source/CameraComponentController.cpp

```diff
@@ -412,6 +415,66 @@ namespace Camera
         return m_isActiveView;
     }
 
+    namespace Util
+    {
+        AZ::Vector3 GetWorldPosition(const AZ::Vector3& origin, float depth, const AzFramework::CameraState& cameraState)
+        {
+            if (depth == 0.f)
+            {
+                return origin;
+            }
+            else
+            {
+                const AZ::Vector3 rayDirection = cameraState.m_orthographic ? cameraState.m_forward : (origin - cameraState.m_position);
+                return origin + (rayDirection.GetNormalized() * depth);
+            }
+        }
+    }
+
+    AZ::Vector3 CameraComponentController::ScreenToWorld(const AZ::Vector2& screenPosition, float depth)
+    {
+        const AzFramework::ScreenPoint point{ static_cast<int>(screenPosition.GetX()), static_cast<int>(screenPosition.GetY()) };
+        const AzFramework::CameraState& cameraState = GetCameraState();
+        const AZ::Vector3 origin = AzFramework::ScreenToWorld(point, cameraState);
+        return Util::GetWorldPosition(origin, depth, cameraState);
+    }
+
+    AZ::Vector3 CameraComponentController::ScreenNdcToWorld(const AZ::Vector2& screenNdcPosition, float depth)
+    {
+        const AzFramework::CameraState& cameraState = GetCameraState();
+        const AZ::Vector3 origin = AzFramework::ScreenNdcToWorld(screenNdcPosition, AzFramework::InverseCameraView(cameraState), AzFramework::InverseCameraProjection(cameraState));
+        return Util::GetWorldPosition(origin, depth, cameraState);
+    }
+
+    AZ::Vector2 CameraComponentController::WorldToScreenNdc(const AZ::Vector3& worldPosition)
+    {
+        const AzFramework::CameraState& cameraState = GetCameraState();
+        const AZ::Vector3 screenPosition = AzFramework::WorldToScreenNdc(worldPosition, AzFramework::CameraView(cameraState), AzFramework::CameraProjection(cameraState));
```

Comment:
suggestion: Rather than using `CameraState` there are overloads for `WorldToScreen` that accept a view and a projection so it might be possible to just pass in `m_atomCamera->GetCameraTransform()` and `m_atomCamera->GetViewToClipMatrix()` directly which will save quite a bit of work - could be a future optimization